### ISSUE TITLE
podman build: error opening file `.../cgroup.freeze`: No such device

### ISSUE THREAD

**Issue by edsantiago** [containers/libpod#5475]
In [CI](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5756523582324736/html/system_test.log.html). Rootless. I don't know which Fedora:
```
# $ ./bin/podman build -t build_test --format=docker /tmp/podman_bats.JIkSvM/build-test
...
# STEP 3: RUN echo mYNEUygNEKRSRisDZZ2w1dztQbAfn5p96CBHIKfkL9GHAcSfum > /8stfSssoTINVW8YZTQ58
# error opening file `/sys/fs/cgroup//user.slice/user-14802.slice/user@14802.service/buildah-buildah631812000-99184.scope/cgroup.freeze`: No such device
# kill container: No such process
# error running container: error reading container state: exit status 1
# Error: error building at STEP "RUN echo mYNEUygNEKRSRisDZZ2w1dztQbAfn5p96CBHIKfkL9GHAcSfum > /8stfSssoTINVW8YZTQ58": error while running runtime: exit status 1
# [ rc=125 (** EXPECTED 0 **) ]
```
CI links: [cirrus](https://cirrus-ci.com/task/5756523582324736) -- [highlighted](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5756523582324736/html/system_test.log.html)

I've restarted test, on the assumption that it's a flake, but want to start tracking regardless.

**Comment by mheon** [comment]
Rootless is on F30, I thiiiiiink?

@cevich can you confirm?

**Comment by edsantiago** [comment]
Yep, it was a flake.

**Comment by edsantiago** [comment]
Oh dear. Seeing it [in ginkgo](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/4808011696832512/html/integration_test.log.html#t--podman-rmi-with-cached-images--1) now (as opposed to BATS the first time). Again, rootless.

**Comment by giuseppe** [comment]
it looks like we are using cgroup v2 on a kernel that doesn't support the freezer control.

We should either use cgroup v1 in this case, or an updated kernel

**Comment by edsantiago** [comment]
But it's a flake: restarting the failed test makes everything OK.

**Comment by rhatdan** [comment]
Are we always running the same kernel?

**Comment by edsantiago** [comment]
In one of the cases above, the `package_versions` step shows `5.5.5-200.fc31.x86_64` for both the [failed](https://cirrus-ci.com/task/5756523582324736?command=package_versions#L0) and [successful](https://cirrus-ci.com/task/6225340670410752?command=package_versions#L0) runs.

**Comment by cevich** [comment]
> Are we always running the same kernel?

Yes, and in general/as much as possible automation avoids updating/changing as much as possible from one run to the next.  IIRC, the crun package is the only exception currently, and only for F31 VMs.

Otherwise (for F30 and Ubuntu), updated packages and kernel requires building new VM cache images (as documented).

**Comment by rhatdan** [comment]
Since we are updating to run against newer kernels going to close this issue.

**Comment by edsantiago** [comment]
It's [back](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/6494953910042624/html/system_test.log.html)
```
not ok 35 podman build - basic test
...
$ podman build -t build_test --format=docker /tmp/podman_bats.Nfkrh8/build-test
...
STEP 3: RUN echo P7P7OoAPTPb03b2fgkfyVDRp65bhJP6FpGECwhdYOZpKPzuAig > /wKvZH6BbgiOoBaO0S6ht
 error opening file `/sys/fs/cgroup//system.slice/buildah-buildah158633376-456211.scope/cgroup.freeze`: No such file or directory
 kill container: No such process
 error running container: error reading container state: exit status 1
 Error: error building at STEP "RUN echo P7P7OoAPTPb03b2fgkfyVDRp65bhJP6FpGECwhdYOZpKPzuAig > /wKvZH6BbgiOoBaO0S6ht": error while running runtime: exit status 1
 [ rc=125 (** EXPECTED 0 **) ]
```
Fedora 31, kernel 5.6.15-200.fc31.x86_64

**Comment by vrothberg** [comment]
@giuseppe @nalind could this a race?

**Comment by giuseppe** [comment]
possibly, I think it could be caused by the cgroup disappearing (systemd cleaning it up) in the middle of accessing it

**Comment by edsantiago** [comment]
Two more: one on [f31](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5397242511097856/html/system_test.log.html) and one on [f32](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/4834292557676544/html/system_test.log.html)

**Comment by edsantiago** [comment]
Another one, [f32](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/4766604812615680/html/system_test.log.html), this time on Dan's PR #6570. Shall I just remove the `podman build` tests from CI?

**Comment by edsantiago** [comment]
Another one, [f31](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5728986053476352/html/system_test.log.html), again in PR #6570, this time in the `healthcheck` test because it too runs `podman build`.

**Comment by edsantiago** [comment]
Two more recent ones: PR #6791 ([log](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5097572006100992/html/system_test.log.html)) and PR #6823 ([log](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/4963661804470272/html/system_test.log.html))

Oh, and two more today: PR #6773 ([log](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/6047533845708800/html/system_test.log.html)) and PR #6821 ([log](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/6468576871710720/html/system_test.log.html))

All failures are only on f31.

Could we please give this one some attention?
